src/suffix.c

```diff
@@ -279,6 +439,57 @@ arrayof(char**) GetList_SuffixTrieMap(TrieMap *trie, const char *str, uint32_t l
   }
 }
 
+static arrayof(char*) _getWildcardArray(TrieMap *trie, const char *str, uint32_t slen, size_t idx, size_t idxlen, struct timespec timeout) {
```

Comment:
Please document the parameters and what each parameter means?

Reply:
I think its getting complicated, you get the original str, the index on which it is best to do the wildcard search, its len. You also assume its NULL terminated. Can we split it to 2 functions? One that will get the chosen substring to search and one then will do the post filtering on the entire searched string?

If I got it completely wrong, I am sorry :)

Reply:
You are correct.
Rewrote the function.